crates/store/src/state.rs

```diff
@@ -397,6 +403,51 @@ impl State {
         self.db.select_notes_by_id(note_ids).await
     }
 
+    /// Queries all the note inclusion proofs matching a certain Note IDs from the database.
+    pub async fn get_note_inclusion_proofs(
+        &self,
+        note_ids: Vec<NoteId>,
+    ) -> Result<BTreeMap<NoteId, NoteInclusionProof>, DatabaseError> {
+        self.db.select_block_note_inclusion_proofs(note_ids).await
+    }
+
+    pub async fn get_block_mmr_proofs(
+        &self,
+        blocks: Vec<BlockNumber>,
+    ) -> Result<Vec<(BlockNumber, MerklePath, RpoDigest, RpoDigest)>, MmrError> {
+        // Grab the merkle paths from the inner state.
+        //
+        // NOTE: Scoped block to automatelically drop the mutex guard asap.
+        //
+        // We also avoid accessing the db in the block as this would delay
+        // dropping the guard.
+        let merkle_paths = {
+            let state = self.inner.read().await;
+            let forest = state.chain_mmr.forest();
+
+            blocks
+                .iter()
+                .map(|block_num| {
+                    let proof = state.chain_mmr.open(todo!(), todo!())?.merkle_path;
```

Comment:
How do I convert an absolute index into a tree + offset index as required by `open`?

My brain tells me this is simple; but for some reason I'm glitching here. Probably just some modulo variation? Or does one have to go bit searching?

Reply:
Opened https://github.com/0xPolygonMiden/crypto/pull/324 to improve this in the future.